sql/sql_utils.cc

```diff
@@ -38,6 +43,65 @@ std::string escapeSql(const char * str)
     return escapeSql(std::string(str));
 }
 
+bool matchSqlFilter(const Utf8String& valueString, const Utf8String& filterString)
+{
+    if (filterString.empty() || valueString.empty())
+        return false;
+
+    Utf8String regExFilter;
+
+    for (const auto& filterChar : filterString) {
+
+        switch (filterChar) {
+            case ('%'): {
+                regExFilter += ".*";
+                break;
+            }
+            case ('_'): {
+                regExFilter += ".";
+                break;
+            }
+            case ('*'): {
+                regExFilter += "[*]";
+                break;
+            }
+            case ('['): {
+                regExFilter += "\\[";
+                break;
+            }
+            case (']'): {
+                regExFilter += "\\]";
+                break;
+            }
+            case ('.'): {
+                regExFilter += "[.]";
+                break;
+            }
+            case ('|'): {
+                regExFilter += "[|]";
+                break;
+            }
+            case ('('): {
+                regExFilter += "[(]";
+                break;
+            }  
+            case (')'): {
+                regExFilter += "[)]";
+                break;
+            }  
+            default: {
+                regExFilter += filterChar;
+            }
+        }
+    }
+
+    //std::cerr << "sql filter regex filter: " << regExFilter << std::endl;
+    boost::u32regex regex = boost::make_u32regex(regExFilter.rawData());
```

Comment:
I thought it was and this seems to be confirmed here:
http://www.boost.org/doc/libs/1_34_1/libs/regex/doc/thread_safety.html.

In any case, this could be part of an optimization round and does not need to be done now I feel.